Hymns/Audio Player/AudioPlayerView.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+import AVFoundation
+import Combine
+
+enum PlaybackState: Int {
+    case waitingForSelection
+    case buffering
+    case playing
+}
+
+struct AudioView: View {
+
+    let player = AVPlayer()
+    let item: URL?
+    fileprivate let seekDuration: Float64 = 5
+    fileprivate var playerCurrentTime: Float64 {
+        CMTimeGetSeconds(self.player.currentTime())
+    }
+
+    @State var currentlyPlaying = true
+    @State var playbackState: PlaybackState = .waitingForSelection
+
+    var body: some View {
+        VStack {
+            AudioPlayerControlsView(player: player,
+                                    timeObserver: PlayerTimeObserver(player: player),
+                                    itemObserver: PlayerItemObserver(player: player), playbackState: $playbackState)
+            HStack(spacing: 30) {
+                //Reset button currently needed for when you tab to chords, piano, or guitar....
+                Button(action: {
+                    if self.currentlyPlaying {
+                        self.currentlyPlaying.toggle()
+                    }
+                    self.playbackState = .buffering
+                    self.player.replaceCurrentItem(with: nil)
+                    guard let url = self.item else {
+                        return
+                    }
+                    let playerItem = AVPlayerItem(url: url)
```

Comment:
Yeah it looks like hymnal.net urls all have 301 redirects before you can get to the mp3 file.

If possible, I'd like that we find some way to get the mp3 file from the 301 redirect (versus hardcoding them), and then it should all just work with the song duration

There's this that might help, but also might not: https://developer.apple.com/documentation/foundation/nsurlsessiontaskdelegate/1411626-urlsession

Reply:
I'm gonna need your help on this one... Is it something we can add to the API?

Reply:
hmm okay. Yeah the slider doesn't work too well atm, but I think we can still submit this and continue to iterate on it